src/metabase/api/dashboard.clj

```diff
@@ -654,6 +621,47 @@
    [:id   ms/Int]
    [:name ms/NonBlankString]])
 
+(defn- do-cards-and-tabs-updates-tracking!
```

Comment:
Why did you have to do this refactor to split out the tracking from the updates? It's not immediately obvious to me

Reply:
I thought it'll be nice to refactor this because this PR adds 3 new events type, which is half of how many events "could" be fired on a single update.

and I think going forward we should follow this pattern, to me it has 2 advantages:
- the tracking is done after the transaction succeed
- readability, it's quite hard to trace what will be tracked in the current form IMO.


but if you find it's out of scope of this PR then I can do the refactor in a separated PR.